### PR TITLE
Test case and fix to NH-3898

### PR DESCRIPTION
**Pull request by hailtondecastro**
Test case and fix to [NH-3898](https://nhibernate.jira.com/browse/NH-3898)
Configuring a property with generated="insert" it makes
"Property.IsUpdatable" "false" even using update="true" in the xml
mapping file.
Please consider keep this issue as Critical, as it does not break the
application, it makes the applications migrated from NHibernate version
2 to 3 or 4 generates inconsistent data within the database.

### CODE REVIEW

**Comment by fredericDelaporte** [comment]
If this patch get merged, it would be probably more likely to happen on master or 4.1 branch.  If you port your PR on master, now it requires VS2017 (Community edition is enough).

The change looks legit, but you need to follow [contributing guidelines](https://github.com/nhibernate/nhibernate-core/blob/master/CONTRIBUTING.md). Notably, use tabs instead of spacing.

**Comment by fredericDelaporte** [comment]
Re-based for fixing the release build, small additional fix added to test.

File: src/NHibernate.Test/NHSpecificTest/NH3898/Fixture.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+			}
+
+			using (var session = OpenSession())
+			using (var tx = session.BeginTransaction())
+			{
+				var employee = session.Get<Employee>(id);
+				Assert.That(employee.PromotionCount, Is.EqualTo(0));
+				employee.Name = "Employee 1 changed";
+				employee.PromotionCount++;
+				Assert.That(employee.PromotionCount, Is.EqualTo(1));
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH3898/Fixture.cs]: That asert does not make sense anymore.

**Comment by hazzik** [comment]
Thanks

**Pull request merged**